docs/hive-metrics.md

```diff
@@ -0,0 +1,108 @@
+<!-- START doctoc generated TOC please keep comment here to allow auto update -->
+<!-- DON'T EDIT THIS SECTION, INSTEAD RE-RUN doctoc TO UPDATE -->
+**Table of Contents**  *generated with [DocToc](https://github.com/thlorenz/doctoc)*
+
+- [Hive Metrics](#hive-metrics)
+    - [Optional Metrics](#optional-metrics)
+    - [Metrics with Optional Cluster Deployment labels](#metrics-with-optional-cluster-deployment-labels)
+    - [List of all Hive metrics](#list-of-all-hive-metrics)
+      - [ClusterDeployment controller metrics](#clusterdeployment-controller-metrics)
+      - [ClusterProvision controller metrics](#clusterprovision-controller-metrics)
+      - [ClusterPool controller metrics](#clusterpool-controller-metrics)
+      - [Metrics controller metrics](#metrics-controller-metrics)
+
+<!-- END doctoc generated TOC please keep comment here to allow auto update -->
+
+# Hive Metrics
+
+Hive publishes metrics, which can help admins to monitor the Hive operations. Most of these metrics are always published, few are [optional](#metrics-with-optional-cluster-deployment-labels).
+
+### Exporting Hive Metrics
+`HiveConfig.Spec.ExportMetrics` specifies whether the operator should enable metrics for hive controllers to be extracted for the prometheus instance running on the cluster. When set to true, `openshift.io/cluster-monitoring` label is applied and  the operator deploys ServiceMonitors so that the prometheus instances can extract metrics. The operator also sets up RBAC in the TargetNamespace so that openshift prometheus in the cluster can list/access objects required to pull metrics.
+
```

Comment:
Is this part clear about what's the default behaviour vs exporting? I wanted to incorporate the learnings from the recent slack discussions around this

Reply:
Maybe we want to defer cleaning this up until we have a better understanding. Based on the (partial) research I've just done, I think the label is only relevant if your `targetNamespace` has an `openshift-` prefix; and we lay the ServiceMonitor and RBAC stuff down as a convenience. However, based on DPTP's experience (the slack discussions you allude to) something seems to be wrong with this picture.